test/cli_test.py

```diff
@@ -350,6 +369,146 @@ def test_unkown_collection(self):
+                "album_art_maxwidth": None,
+                "album_art_format": None,
+                "album_art_deinterlace": False,
+                "album_art_quality": 0,
+            }
+        }
+        self.external_config = self.config["alternatives"]["myexternal"]
+
+    def test_resize_art(self, tmp_path: Path):
+        def assert_art_size(path: bytes):
+            size = ArtResizer.shared.get_size(path_in=path)
+            assert size is not None
+            assert size[0] == 1  # width
+            assert size[1] < 3  # height
+
+        album = self.add_album(myexternal="true")
+        album.store()
+        self.runcli("alt", "update", "myexternal")
+
+        external_album_path = self.external_dir / "artist 1" / "album 1"
+        external_art_path_bytes = bytes(str(external_album_path / "COVER.png"), "utf8")
+
+        self.external_config["album_art_copy"] = True
+        self.external_config["album_art_maxwidth"] = 1
+        album.set_art(self.IMAGE_FIXTURE1)
+        album.store()
+        self.runcli("alt", "update", "myexternal")
+        assert_art_size(external_art_path_bytes)
+        assert ArtResizer.shared.get_format(path_in=external_art_path_bytes) == "PNG"
+
+        self.external_config["album_art_format"] = "JPEG"
+        self.runcli("alt", "update", "myexternal")
+
+        external_art_path = external_album_path / "COVER.jpg"
+        external_art_path_bytes = bytes(str(external_art_path), "utf8")
+
+        assert_art_size(external_art_path_bytes)
+        assert ArtResizer.shared.get_format(path_in=external_art_path_bytes) == "JPEG"
+        assert external_art_path.name == "COVER.jpg"
+        # Check that original album art is still around to verify that
+        # the reformat was not done in-place
+        assert album.artpath
+        assert Path(str(album.artpath, "utf8")).is_file()
+
+        # Test that reformat is idempotent
+        mtime_1 = external_art_path.stat().st_mtime
+        self.runcli("alt", "update", "myexternal")
+        mtime_2 = external_art_path.stat().st_mtime
+        assert mtime_1 == mtime_2
+
+    def test_copy_art(self):
+        # Initially add album without artwork. Do not do resizing
+        self.external_config["album_art_embed"] = False
+        self.external_config["album_art_copy"] = True
+
+        album = self.add_album(myexternal="true")
+        album.store()
+        self.runcli("alt", "update", "myexternal")
+
+        external_album_path = self.external_dir / "artist 1" / "album 1"
+        external_art_path = external_album_path / "COVER.png"
+
+        # Test that no artwork is placed
+        self.runcli("alt", "update", "myexternal")
+        assert not external_art_path.is_file()
+
+        album.set_art(self.IMAGE_FIXTURE1)
+        album.store()
+        self.runcli("alt", "update", "myexternal")
+        assert_same_file_content(external_art_path, self.IMAGE_FIXTURE1)
+
+        # Update art file
+        album.set_art(self.IMAGE_FIXTURE2)
```

Comment:
I guess I was to optimistic about removing `touch_art`. There were some build failures

```suggestion
        album.set_art(self.IMAGE_FIXTURE2)
        touch_art(album.artpath, external_art_path)
```